src/decimal.rs

```diff
@@ -1514,6 +1514,122 @@ impl Decimal {
             flags: flags(self.is_sign_negative(), quotient_scale),
         })
     }
+
+    /// The estimated exponential function, e<sup>x</sup>, rounded to 8 decimal places. Stops
+    /// calculating when it is within tolerance is roughly 0.000002 in order to prevent
+    /// multiplication overflow.
+    fn exp(&self) -> Decimal {
+        if self == &Decimal::zero() {
+            return Decimal::one();
+        }
+
+        let tolerance = Decimal::new(2, 7);
+        let mut term = self.clone();
+        let mut result = self.clone() + Decimal::one();
+        let mut prev_result = Decimal::zero();
+        let mut factorial = Decimal::one();
+        let mut n = Decimal::new(2, 0);
+
+        // Needs rounding because multiplication overflows otherwise.
+        while (result - prev_result).abs() > tolerance && n < Decimal::new(24, 0) {
+            prev_result = result;
+            term = self * term.round_dp(8);
+            factorial *= n;
+            result += (term / factorial).round_dp(8);
+            n += Decimal::one();
+        }
+
+        result
+    }
+
+    /// Raise self to the given unsigned integer exponent: x<sup>y</sup>
+    fn powi(&self, exp: u64) -> Decimal {
+        if exp == 0 {
+            Decimal::one()
+        } else if exp == 1 {
+            self.clone()
+        } else {
+            let mut result = self.clone();
+            for _ in 1..exp {
```

Comment:
There are some optimizations we can look into here potentially.

1. The easiest one is to reduce the number of calculations required. e.g. `3^9` could be broken down into `(((3^2)^2)^2)*3`. It's subtle, but instead of `9` multiplications, it would only do `4`. 
2. The second is to avoid the boxing/unboxing each loop by using an intermediary format.

Some experimentation was done a while ago on [this here](https://github.com/paupino/rust-decimal/compare/feature/pow#diff-21e0bf62805e3ff1074a826a701e3e419044270d9c8b2c9cbf4090d519c8da0bR2754).

I quite like the `powi` version though as it makes it clear and straight forward to what is required.

Anyway, probably not necessarily something to hold this up but just adding comment.

Reply:
I went ahead and updated the calculation so it works like you suggested. Looks like it makes it about 33% faster.
`654 ns/iter (+/- 23)` from 993